src/forms/BoundForm.tsx

```diff
@@ -28,8 +28,13 @@ import { FormLines } from "./FormLines";
 type BoundFieldInputFnReturn = { component: ReactNode; minWith: string };
 type BoundFieldInputFn<F> = (field: ObjectState<F>[keyof F]) => BoundFieldInputFnReturn;
 
-type BoundFormRowInputs<F> = {
-  [K in keyof F]: BoundFieldInputFn<F> | ReactNode;
+const reactNodePrefix = "reactNode" as const;
+type CustomReactNodeKey = `${typeof reactNodePrefix}${string}`;
```

Comment:
I suspect you'd be able to create something like

```ts
type CustomReactNodeKey<T extends string> = `${typeof reactNodePrefix}${T}`;
```

then maybe that opens the door for BoundFormRowInputs to generate the type

```ts
// given some config `{ firstName: string }`
// inputs become: `{ firstName: BoundFieldInput<F>, firstNameReactNode?: ReactNode }`
```

Reply:
Sounds good, ended up giving that a try and having the enumerated list does help with discoverability of the feature, given we still allow any `reactNode*` prefixed key as valid for when we need custom JSX that doesn't live in the form at all. 

<img width="1104" alt="image" src="https://github.com/user-attachments/assets/5fa49dd6-8376-4297-934d-52afa10ae1dd" />


<img width="812" alt="image" src="https://github.com/user-attachments/assets/163754f6-a94e-4d9f-b36b-f25d9d4e72dc" />